crmsh/xmlutil.py

```diff
@@ -1482,4 +1482,68 @@ def attributes(typename, rules, values, xmlid=None, score=None):
     return e
 
 
+class CrmMonXmlParser(object):
+    """
+    Class to parse xml output of crm_mon
+    """
+    def __init__(self, peer=None):
+        """
+        Init function
+        when peer set, parse peer node's results
+        """
+        self.xml_elem = None
+        self.peer = peer
+
+    def _load(self):
+        """
+        Load xml output of crm_mon
+        """
+        _, output, _ = get_stdout_stderr(constants.CRM_MON_XML_OUTPUT, remote=self.peer)
+        self.xml_elem = text2elem(output)
+
+    @classmethod
+    def is_node_online(cls, node):
+        """
+        Check if node online
+        """
+        cls_inst = cls()
+        cls_inst._load()
+        elem_list = cls_inst.xml_elem.xpath('//node[@name="{}" and @online="true"]'.format(node))
+        return len(elem_list) != 0
+
+    @classmethod
+    def is_resource_configured(cls, ra_type, peer=None):
+        """
+        Check if the RA configured
+        """
+        cls_inst = cls(peer=peer)
+        cls_inst._load()
+        elem_list = cls_inst.xml_elem.xpath('//resource[@resource_agent="{}"]'.format(ra_type))
+        return len(elem_list) != 0
+
+    @classmethod
+    def has_resource_running(cls, peer=None):
```

Comment:
You may like the consistent naming style with is_any_resource_running(), instead of has_resource_running() ?